moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp

```diff
@@ -587,7 +587,9 @@ bool PlanningSceneMonitor::newPlanningSceneMessage(const moveit_msgs::PlanningSc
 
     if (!scene.is_diff && parent_scene_)
     {
-      // clear maintained (diff) scene_ and set the full new scene in parent_scene_ instead
+      // If there is no new robot_state, transfer RobotState from current scene to parent scene
+      if (scene.robot_state.joint_state.name.empty() && scene.robot_state.multi_dof_joint_state.joint_names.empty())
```

Comment:
```suggestion
      if (scene.robot_state.is_diff || (scene.robot_state.joint_state.name.empty() && scene.robot_state.multi_dof_joint_state.joint_names.empty()))
```

Ideally I think `is_diff` should be enough because if the state is not a diff it should include the full joint state and otherwise trigger [the spurious error "Found empty JointState"](https://github.com/moveit/moveit/blob/3aed1dc85de2f68ffcdeff1250f2ba8ec34eb20c/moveit_core/robot_state/src/conversions.cpp#L357-L361).
But given the wide interpretation range of these flags over the years it makes sense to catch the case where `is_diff==false` and the joints are still empty as well.

Reply:
You are right. `is_diff` should be sufficient. At least [MTC correctly sets the flag](
https://github.com/moveit/moveit_task_constructor/blob/2d0244b921a4149e2de5c9019658ad881a3bb2bb/capabilities/src/execute_task_solution_capability.cpp#L176).